src/adagrams.js

```diff
@@ -1,7 +1,104 @@
+const _ = require('lodash');
+
 const Adagrams = {
+  letters: ['A', 'A', 'A', 'A', 'A', 'A', 'A', 'A', 'A',
+    'B', 'B',
+    'C', 'C',
+    'D', 'D', 'D', 'D',
+    'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E',
+    'F', 'F',
+    'G', 'G', 'G',
+    'H', 'H',
+    'I', 'I', 'I', 'I', 'I', 'I', 'I', 'I', 'I',
+    'J',
+    'K',
+    'L', 'L', 'L', 'L',
+    'M', 'M',
+    'N', 'N', 'N', 'N', 'N', 'N',
+    'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 
+    'P', 'P', 
+    'Q',
+    'R', 'R', 'R', 'R', 'R', 'R',
+    'S', 'S', 'S', 'S',
+    'T', 'T', 'T', 'T', 'T', 'T',
+    'U', 'U', 'U', 'U',
+    'V', 'V',
+    'W', 'W',
+    'X',
+    'Y', 'Y',
+    'Z'],
+  scores: {
+    1: ['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T'],
+    2: ['D', 'G'],
+    3: ['B', 'C', 'M', 'P'],
+    4: ['F', 'H', 'V', 'W', 'Y'],
+    5: ['K'],
+    8: ['J', 'X'],
+    10: ['Q', 'Z']
+  },
+
   drawLetters() {
-    // Implement this method for wave 1
+    return _.sampleSize(this.letters, 10);
+  },
+
+  usesAvailableLetters(input, lettersInHand) {
+    const playedLetters = (input.toUpperCase()).split('');
+    let valid = {};
+    let result = true;
+
+    lettersInHand.forEach((letter) => {
+      valid[letter] ? valid[letter] += 1 : valid[letter] = 1;
+    });
+
+    playedLetters.forEach((letter) => {
+      if (valid[letter]) {
+        valid[letter] <= 0 ? result = false : valid[letter] -= 1;
+      } else {
+        result = false;
+      }
+    });
+    return result;
   },
+
+  scoreWord(word) {
+    const input = (word.toUpperCase()).split('');
+    const array = Object.entries(this.scores);
+    let total = 0;
+
+    input.forEach((letter) => {
+      array.forEach((score) => {
+        if (score[1].includes(letter)) {
```

Comment:
This ends up reading weirdly (the `score[1]` bit)... There's no obvious way to make this read weirdly without refactoring the whole solution, but consider restructuring `this.scores` so that you won't have to rely on the structure that `Object.entries` gives you